### PR TITLE
[MAINT] make test_percentile_range for canica less brittle

### LINKED ISSUE #3546: flaky `test: test_canica.py::test_percentile_range`
It seems to fail "once in a while"

https://github.com/nilearn/nilearn/actions/runs/4301638495/jobs/7499114761#step:8:16363
https://github.com/nilearn/nilearn/actions/runs/4292758055/jobs/7479594709#step:8:16252
https://github.com/nilearn/nilearn/actions/runs/4277980448/jobs/7447269258#step:8:16124
https://github.com/nilearn/nilearn/actions/runs/4251899059/jobs/7394830022#step:8:16113

```bash
FAILED nilearn/decomposition/tests/test_canica.py::test_percentile_range - assert 2 == 1
 +  where 2 = sum([False, False, False, False, False, False, ...])
```

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3546 

Changes proposed in this pull request:

- simplify test by making sure that only one warning with the expected message is thrown

### Unrelated

tidy up gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,40 +4,48 @@
...
+
 examples/results/
 examples/**/results/
 examples/**/*.pdf
+
+.ropeproject/
+
+tags
+
+tmp
```
Comment (bthirion) [inline .gitignore:51]: All these changes seem unrelated . Shouldn't they go to another PR ?

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

File: .gitignore (hunk 1 above)
Comment (Remi-Gau) [reply .gitignore:51]: yup 
will put that on another branch

Comment (Remi-Gau) [reply .gitignore]: See #3565

**Review by bthirion — APPROVED** [review]
LGTM.

**Review by ymzayek — APPROVED** [review]

**Comment by Remi-Gau** [comment]
merging and hoping this will not resurface (famous last words)
